src/Apps/Settings/Routes/Insights/Components/CareerHighlights/InsightsCareerHighlightCard.tsx

```diff
@@ -46,6 +54,47 @@ export const InsightsCareerHighlightCard: React.FC<InsightsCareerHighlightCardPr
 
         <Text variant={["xs", "sm-display"]}>{label}</Text>
       </Flex>
-    </Flex>
+    </CardWrapper>
   )
 }
+
+// extending ClickableProps since the type already has FlexProps (BoxProps)
+interface CardWrapperProps extends ClickableProps {
+  onClick?: () => void
+}
+
+const CardWrapper: React.FC<CardWrapperProps> = ({
+  onClick,
+  children,
+  ...rest
+}) => {
+  const isCareerHighlightModalEnabled = useFeatureFlag(
+    "my-collection-web-phase-7-career-highlights-modal"
+  )
+
+  if (isCareerHighlightModalEnabled) {
+    return (
+      <ClickableCard onClick={onClick} {...rest}>
+        {children}
+      </ClickableCard>
+    )
+  }
+  return <Card {...(rest as FlexProps)}>{children}</Card>
```

Comment:
I'd avoid doing this casting. It looks like what you want to do is just extend `BoxProps`? Since obviously this could lead to bad things like someone accidentally sticking button props on something that shouldn't be clicked.

If you want to split the props in a type-safe way we have a utility for that: https://github.com/artsy/palette/blob/main/packages/palette/src/utils/splitProps.ts